### DISCUSSION TITLE
canvas size

### DISCUSSION THREAD

**Discussion by Xeltosh** [hako-mikan/sd-webui-regional-prompter#163] [Q&A]
is it possible to modify the canvas-size? i am experimenting with some stuff and would like to test larger images, but the canvas is capped at 1024x1024 pixels....

**Comment by Symbiomatrix** [comment]
1) Actually, it's supposed to be 2048x2048 (see canvas_width in rp.py). I'm not sure why it's capping at 1024, haven't noticed it till now. My best guess is that it's sharing the same components as controlnet's canvas dimensions on gradio's side, due to the same name.
Edit: Actually, it's probably because ui-config.json stores component settings under their label designation rather than a unique id (yuck), so changing "canvas" to "rp" or something should fix this. Done, should be updated now.
2) The dimension specification actually doesn't matter as much as the proportion; the mask is resized during processing according to necessity. Unless you're drawing to single pixel level accuracy, you wouldn't notice the difference, and even then you probably wouldn't since that'd be like trimming a hair off a gorilla on the grand scale of the generation process.
